### DISCUSSION TITLE
Pasture implementation in FATES

### DISCUSSION THREAD

**Discussion by rosiealice** [NGEET/fates#936] [Ideas]
Matvey Debolskiy (who has a project with Hanna Lee looking at grazing impacts in Mongolia) and I had a discussion last week about potential implementations of pasture and grazing in the CLM/FATES universe. @billsacks mentioned he may have some resources to think about this also, and there are many people with a stake in these ideas, so I'm moving the discussion here to avoid email thread purgatory. 

In principle, FATES should have the capability to represent pasture in a fairly natural fashion, as grazing (as the removal of leaf and stem biomass, and the elevated mortality of trees) can be handled quite logically by either the new crown damage capabilities, and/or the native fates mortality and disturbance routines, (depending on how you think about it). This is generally more challenging to do in big leaf mode as, for example, re-growth after grazing is not going to have any storage pools, the LAI algorithms for deciduous plants are not very flexible, and plants will easily get off their (unspecified) allometry and e.g. generate huge respiring root pools that will likely kill them off. 

At the moment, we have no pasture or crop representations in FATES. Land use presently works by taking the harvest rates from LUH2 and converting these into disturbance rates in FATES, which generate new types of patch that have a 'secondary forest' label, and do not fuse back into the primary forest patches. @sshu88 in particular has been testing this implementation extensively in recent months. 

We discussed previously (e.g, with @ckoven ) expanding this logic so that some of the harvest instead creates crop patches that have their own label/flag, to mirror how we currently track primary and secondary forest, and enable an order 0 crop->nat veg transition to occur and form a platform for more crop physiology and phenology development.

In light of the pasture and grazing discussion, One approach to this would thus be to add pasture (as well as crop) flags, and have the 'grazing' flux drive their creation. 

This would be a good first step, and would allow us to e.g. test grazing algorithms and how the affect LAI, biomass, etc. 

It isn't, however, really where we want to end up, because:

a) the different land uses would all be on the same column and share the same soil and litter. This limits how we can look at the impacts of land use on column based processes including nutrients, water, snow and fire. 

b) we don't currently have the full set of LUH2 transitions read into FATES (or either HLM, right?) so we can't partition the gross changes appropriately. (I am a bit hazy, to be honest, on how the existing scheme will differ from LUH2-land. If someone else has a better way of articulating this that would be useful). 

These two issues are somewhat orthogonal to each other and can be addressed separately.

I'm going to stop here as I've run out of time, but wanted to kick this off to see whether I've captured the state of play accurately if there are any other threads on this area we should integrate, and also to figure out if there are any good synergies with what @billsacks and @dlawrenncar are planning that we could use to accelerate this. 

Noting that no one has actually decided to do any of these things yet. If we can somehow divide and conquer, that would be ideal...

**Comment by jkshuman** [comment]
Thanks for bringing this up @rosiealice This has relevance to many groups and people, and I appreciate you creating this discussion to track the conversation. 

tagging @dlawrenncar @billsacks @danicalombardozzi @jenniferholm @sshu88 @glemieux @rgknox @ckoven and Sam Rabin

**Comment by samsrabin** [comment]
Thanks for this writeup, Rosie! I'm being funded for a few months' worth of software engineering time, so I may be able to contribute. Looking forward to further discussion.

**Reply by rosiealice** [reply]
Thanks Sam, that's great to know. 

We were planning to talk about this at the CLM meeting on the 17th Nov at 1pm MDT if you are available? (Or indeed, anyone else reading this thread who is interested in this issue. Let @wwieder know if you don't have the link.)

**Comment by jkshuman** [comment]
Correction that we finding a new time for this meeting, and will post when that is decided. Contact @wwieder for link details.

**Comment by mvdebolskiy** [comment]
Thanks  @rosiealice,


>a) the different land uses would all be on the same column and share the same soil and litter. This limits how we can look at the impacts of land use on column-based processes including nutrients, water, snow, and fire.

I am not sure if this is an issue. If pasture is dealt with in the same way other CFTs are in CLM -- it will get a separate soil column, however, I am not sure if fire works on CFTs. On the other hand, if rangeland is set as a natural PFT (which it kind of is in reality) it will end up on the nat.veg. soil column.  Same for the natural herbivores (I assume it is desirable to consider them for the future).

**Reply by rosiealice** [reply]
Hi @mvdebolskiy, 

So, in FATES at present, all vegetation lives on the same column, and we don't as yet have the capability to give a new patch it's own soil column. Adding that capability would be probably the primary piece of development to get us where we want to be, but it is not, as I understand, a trivial thing...

**Reply by mvdebolskiy** [reply]
But you don't need to do that. Since there are separate columns for each CFT in Crop landunit in CLM. There is also [hillslope hydrology pr](https://github.com/ESCOMP/CTSM/pull/1715) that allows additional columns with different PFTs for the naturally vegetated land unit. So, I suppose that the simplest solution would be extending the clm-fates interface to allow for more land units/columns to be passed. I have no idea how it works in E3SM though.

Also, more columns will increase cpuhours/modelyears per day, so I would consider keeping column splitting to the minimum level (but that should be tested).

**Reply by ckoven** [reply]
Hi @mvdebolskiy the issue is that the top-level FATES data structure (a "site") corresponds to a CLM/ELM column.  In a static sense, CLM/ELM could initialize different FATES sites on different columns (which could be on different land units or topo units, etc...), and pass an initial patch label to each FATES site to initialize as when it does so (currently FATES implicitly initializes all patches as having a `primaryforest` label). So then if FATES had pasture management practices represented on pasture-labeled patches, it would  work as I think you are describing.  The harder problem is what to do in a transient sense.  Basically in addition to the above there would need to be some sort of patch-transfer data structure that carved off some patch area and its associated properties (cohorts, litter, etc) from one fates site, passed it to another fates site somewhere on the same gridcell, and triggered an associated change in the underlying column (and possibly also landunit) areas that corresponded to those two sites.  If we had that, it would solve lots of these problems. As @rosiealice says, it is on the long-term to-do list.

**Reply by billsacks** [reply]
> there would need to be some sort of patch-transfer data structure that carved off some patch area and its associated properties (cohorts, litter, etc) from one fates site, passed it to another fates site somewhere on the same gridcell, and triggered an associated change in the underlying column (and possibly also landunit) areas that corresponded to those two sites

I'm looking forward to talking more about this. Conceptually, what you're describing sounds similar to what we already have in place in CTSM (and I'm pretty sure ELM, too) for handling changes in column-level state variables when there are changes in column areas, so I wonder if we can get away with letting the host land model handle (many of) these changes. But I think we'll need to talk more to understand the details.

**Reply by rosiealice** [reply]
I guess it's maybe worth noting that when FATES makes a new patch now, it carves off some of its cohorts, rearranges all their densities and pointers, and reassigns them elsewhere. 

Pulling off a piece of the FATES column (nat veg land unit) and putting it on the crop column (& land unit) seems harder to me, but that's maybe because I've always been slightly scared of the dynamic landunits problem. 

Perhaps (to get to the goal where e have crop separate column functionality equivalent to clm5) we "just" need a way of marrying fates "spawn patches" routine/logic with the existing dynamic land units code?

**Reply by billsacks** [reply]
I guess I'll want to better understand the needs here. Naively I don't understand why you'd need to migrate cohorts when transitioning between natural veg and crop, because it seems like the new area will start from bare ground, but I'm probably missing something due to my lack of understanding of FATES.

I'm not sure if this helps at all, but for non-FATES landuse changes, we separate the patch-level (vegetation) dynamics from the column-level (soil) dynamics. The first step involves adjusting the patch-level variables: a growing patch gets some new area that effectively starts bare (in the non-FATES world this simply involves decreasing the per-area carbon and nitrogen density; in FATES this would be handled much better); a shrinking patch keeps the same per-area carbon and nitrogen density but sends the "harvested" carbon and nitrogen to the soil or atmosphere. Then (after incorporating any harvested C & N from the first step) we adjust the column-level variables to conserve soil C & N. (This is described in more detail in the [CLM tech note](https://escomp.github.io/ctsm-docs/versions/master/html/tech_note/Transient_Landcover/CLM50_Tech_Note_Transient_Landcover.html#carbon-and-nitrogen-conservation).) Perhaps naively, I am hoping that FATES can handle the patch-level dynamics while letting the host land model handle the column-level dynamics. I'd like to talk more to understand why my naive hope won't work ;-)

**Reply by ckoven** [reply]
@billsacks it's a good point; I guess we could probably start out at least with the assumption that all disturbances that are intense enough to be represented by column-area shifts are ones that completely clear out the vegetation; if we find cases where that doesn't work (e.g. the fire point below) then we should probably be able to relax that assumption subsequently. But I guess I just want to make two other points here.

The first is just that there is a very distinct problem between what FATES does with pasture (or range, etc) patches in terms of grazing, fire management, harvest, what PFTs are allowed to grow, etc, the logic of which shouldn't really need to care about any of the underlying structures of whether it is on its own column or not.  So I think a lot of science can be done without having to deal with these other questions.

The second point, for the question of how to allow the disturbance dynamics to propagate underground into changing column areas, is that whatever solution we pursue should work also for disturbance that is endogenous, e.g. by fire, and not just for disturbance that is exogenous, i.e. read from the LUH2 transition matrix or similar file.  So I would advocate that the decision for how much area to transfer from one column to another can be made from within FATES and passed outwards from there, so that way it can work for both endogenous and exogenous disturbances, rather than a CLM/ELM logic that is reading files and changing FATES site areas just on the basis of that.  Even if in the short term that means passing the transition matrix file into FATES and then letting it act on that info, I think that will allow for better future compatibility when we went to do things like allow dynamic patch-column correspondence for processes like coupled fire-permafrost successional dynamics in the boreal.

**Reply by rosiealice** [reply]
Concurring with your two points Charlie. 

@billsacks I agree for crops it is true that we would probably somehow want to nuke all the plants, but if we wanted to transition to say, a pasture land unit or column, we might want to keep, say, some of the existing vegetation and let the notional animals affect what happens to it.

I feel like we might need ultimately to split this discussion into these threads, which can be considered rather separate tasks: 
1. Reading in LUH2 transitions 
2. Making order-1 crop and pasture patches in fates
3. Deciding what to do with those patches r.e. management
4. FATES usage of multiple columns and dynamic land units. 
5. Actually representing crops in fates (link to AgSys, etc.) 

Anything else?

**Reply by billsacks** [reply]
Sounds good @ckoven . I don't think there's any assumption in the CLM/ELM dynamic landunit code that ties it to exogenous forcings. I like the idea that FATES would dictate this.

**Reply by billsacks** [reply]
@rosiealice Ah I see your point about transitioning to pasture. The world is a messy place.... (Anyone interested in switching to the problem of cloud physics? It seems a lot simpler....)

**Comment by jkshuman** [comment]
Noting here that the meeting time had to be changed, and will be posted when it is determined. (NOT happening on Nov 17)

**Comment by samsrabin** [comment]
Thanks all for the great discussion in the 9 am (MT) call today. [Here](https://docs.google.com/document/d/1326u7Bex9Rwz3sybmpLM7RaVy0UYKvnYNGJASlTdIrw/edit#) is a link to Will's notes, with a couple of points I added to cover discussion after he had to leave.

One thing I want to add (probably not a hugely important distinction): @rosiealice, I think you're imagining rangeland/pasture creation as happening like secondary forest—there's some amount of harvested material that LUH2 provides, and from there FATES does some logic to get the new secondary forest area. That's not really how LUH2 handles pasture or crops. Instead, those are purely just area inputs that LUH2 provides.

**Comment by evalieungh** [comment]
Chiming in with some naive questions to this very interesting discussion. When you say pasture, are you envisioning grasslands that are mowed, fertilised, irrigated, or just grazed in certain parts of the growing season? Would they need to be on a crop landunit? Could you capture natural grazing as well, e.g. by herds of large herbivores on the Serengeti or historically in Europe/N.-America/Asia? 

In my understanding of Norwegian traditional pastures and rangeland, the ecosystem is created by removing trees (or not, in many cases) and just letting animals graze enough that the vegetation composition changes. E.g. dwarf shrubs and tree seedlings are disadvantaged while herbaceous plants and grasses are pre-adapted and will have an advantage. 

When I heard about initiatives to include grazing, the first thought was that you would copy and modify the logging mechanisms to mimic animals selectively harvesting biomass from leaves or other tasty tissues. It would be great (and more ecologically realistic, I think) if the intensity of grazing could be somehow changeable, to end up with something belonging with the crop landunit and some systems being tree-less but otherwise 'naturally vegetated' and maybe some even with trees but shifting the competitive interactions between different PFTs.

**Reply by samsrabin** [reply]
Very good questions and definitely not naive! So far, we've only discussed livestock, not also natural consumers. The various levels of complexity as you've mentioned for livestock (do we capture pasture vs. rangeland, what sort of management is present in addition to biomass consumption, is consumption intensity variable spatially/temporally, do we also allow livestock _browsing_ [consumption of woody material], etc.) would all be great to include eventually, but of course we need a plan to get there. 

The idea, I think, is to start simple but build features that can support the eventual inclusion of all that complexity.

In our discussion today, I think we decided that if it's decided to use land units in FATES, it would be cleanest to have livestock-grazed land be a separate land unit from cropland and natural vegetation. However, Rosie floated the idea that land units might not be needed at all—instead, there could be one general column type, with land uses and management being denoted by various column-level metadata.

**Comment by jkshuman** [comment]
Adding this Goldewjick et al 2017 link for details on methods of separating grazing land into pasture vs rangeland for HYDE data used in LUH2 (page 935 of manuscript).
https://essd.copernicus.org/articles/9/927/2017/essd-9-927-2017.pdf 

"Whether or not natural vegetation has been converted to establish grazing land is a very relevant question for study- ing the impacts of land use change. While most range- lands occur on land with mostly natural vegetation, low- intensity livestock grazing is also located in former forest or woodland areas, e.g. in Brazil. Therefore, after consultation with the Land Use Model Intercomparison Project (LU- MIP), we recommend that our maps of pasture and range- land be used for modelling purposes in the following way: for pastures, all natural vegetation is cleared and replaced by grass species. For rangeland, the natural vegetation remains intact if it is non-forest, but is cleared if it is forest. HYDE includes this distinction by providing two types of range- land: (1) rangeland-natural is located in non-forest biomes (terrestrial ecoregions; Olson et al., 2001) and is therefore assumed not to have undergone conversion of natural vege- tation. (2) Rangeland-converted is located in forest biomes (terrestrial ecoregions; Olson et al., 2001) and is assumed to have undergone conversion of natural vegetation."

**Comment by billsacks** [comment]
Part of the discussion today (which we should probably spin off into its own thread, but I'll include it here for now) involved how to move to a representation of gross rather than net transitions between land cover types. One thing I'd like to hear thoughts on at some point is: How problematic will it be if CTSM (or ELM) continue to only treat conservation of below-ground (column-level) C & N using net changes in areas, rather than knowing about the full matrix of exactly which areas transitioned to which other areas? This treatment of only net changes – with implicit assumptions that columns with increasing areas are growing equally into all columns with shrinking areas – was a pretty fundamental design decision in the dynamic column/landunit implementation, and reworking this to capture the full transition matrix might require a pretty large overhaul of the dynamic landunit infrastructure.

**Comment by evalieungh** [comment]
Let's say, hypothetically, that someone wants to set up a simple model experiment with the easiest possible grazing implementation in FATES. The basic requirement would be to kill some portion of canopy/leaf biomass, and to remove that biomass from the system (i.e. make sure it doesn't become litter and fertilise the soil). Is that possible with the <https://github.com/NGEET/fates/pull/788> Crown damage module, for instance? It looks like it should be OK to kill off canopy/leaf parts, but where does that biomass go? Is it tracked at all?

Assuming the use is similar to the logging module ([described in user guide](https://fates-users-guide.readthedocs.io/en/latest/user/Running-FATES-with-the-selective-logging-module-activated.html)), could we mimic very simplistic grazing by setting hlm_use_canopy_damage and hlm_use_understory_damage = .true., and modifying fates_damage_frac, fates_damage_mort_p1, fates_damage_mort_p2, and fates_damage_recovery_scalar?

**Reply by ckoven** [reply]
Hi @evalieungh -- my advice would be not to use the damage module for this problem, since the essential feature of that is to track the dynamics of plants that have been damaged separately from those who haven't, by changing their allometries, etc.  So that strikes me as not the simplest solution.  What I'd maybe suggest instead is to write new logic that follows along somewhat analogously to what `PRTBurnLosses` does.  `PRTBurnLosses` removes tissues from plants without the possibility for nutrient retranslation, and adds them to the `prt%variables(i_var)%burned(i_pos)` attribute, which ends up being aggregated into a burn flux and routed away from the decomposition cascade.  One could make a similar attribute called `prt%variables(i_var)%herbivory(i_pos)` or similar, and an analogous subroutine (`PRTHerbivoryLosses` or similar?) that removes some fraction of plant tissues and sends them to the atmosphere. I'm not sure where in the call sequence would be best to call such a subroutine though. `PRTBurnLosses` is part of the disturbance routine (i.e. it only happens on patches as they are being disturbed by fire), I wouldn't think we'd want to treat herbivory as a patch-generating disturbance type, so perhaps triggering it somewhere where normal leaf turnover processes happen would make sense?

**Reply by mvdebolskiy** [reply]
In the [simplest model](http://www.jstor.org/stable/2462920) consumed biomass is not tracked. A percentage of grazed C and N is returned to different pools or the atmosphere as emissions (the coefficients are empirically derived and fixed), and the rest just vanishes into a hungry void. So yes, you can do it in the same manner as harvest (or logging), but you still have to have an excretion flux from animals to the pools. In more [sophisticated](https://www.sciencedirect.com/science/article/abs/pii/S0065211319300719) [models](https://iopscience.iop.org/article/10.1088/1748-9326/aacc75) the output fluxes are still a percentage (or some function) of intake, but the intake depends on the animal energy balance (and sometimes implicit or explicit "internal" variables for an animal) that allows to predict animal weight and milk production f.e.

**Reply by evalieungh** [reply]
Thanks for the advice and literature links, @ckoven and @mvdebolskiy. I'll take a look (and try to find a 'Fortran for dummies' book)!

**Reply by rosiealice** [reply]
Hi all. @evalieungh and I are looking through the logic of adding a grazing function analagous to the burned loss function. 
We think we're mostly on track, but were wondering what, if anything, happens to the site_mass%burn_flux_to_atm flux? Is there a hlm side variable into which this is fed? I/we got a bit lost in the interface trying to figure this out...

**Reply by glemieux** [reply]
I don't think there is.  Looking at the code, it appears that it is used in the `TotalBalanceCheck` to calculate the total flux error.  There is some additional error diagnostic usage in the patch dynamics code, but you have to turn on the hardcoded `debug` switch in the module.

**Reply by evalieungh** [reply]
Tagging @XiulinGao. Rosie and I sketched out a plan for a very simplistic grazing capability for single-site runs. Perhaps this could be relevant for you too since I gathered you work with FATES.

**Reply by mvdebolskiy** [reply]
@evalieungh can you link your branch, where you push your development for this? So we can follow what is happening more easily [Here is mine](https://github.com/mvdebolskiy/CTSM/tree/clm_graz_filt).

**Reply by XiulinGao** [reply]
@evalieungh Thanks! Would like to know more about it if you can point me to the specific branch.

**Reply by evalieungh** [reply]
I haven't started working on it yet, but I'll use this branch: <https://github.com/evalieungh/fates/tree/grazing>. If you had planned some other implementation, Xiulin, maybe we should meet and coordinate to avoid double work or help each other. 
@mvdebolskiy's work seems much more comprehensive so it will likely take longer to complete (unless I'm incredibly slow, which is not impossible). Thanks for linking your branch, Matvey.

**Reply by evalieungh** [reply]
Hi all, there is some progress on the "simplistic grazing" scheme, see [evalieungh/fates grazing branch](https://github.com/evalieungh/fates/tree/grazing). I have copied the PRTBurnLosses subroutine to make PRTGrazeLosses. @rosiealice and I just discussed how to proceed, and we wonder if we should look at PRTMaintTurnover also. Ideally, grazing should:

- be a rate/continuous thing, e.g. as a function of new leaves produced per timestep,
- not a disturbance event (i.e. not cause new patch),
- send grazed biomass to atmosphere/away instead of a litter pool.

@ckoven or @rgknox, do you have any input on this?

**Reply by rgknox** [reply]
@evalieungh , why send grazed biomass to the atmosphere and not the litter pool, or a combination of both? Sorry if the answer is obvious. Is it because grazer's net respire so much?  Also, how about nitrogen and phosphorus grazed from plant matter?  Should we send that to the litter pool?

**Reply by evalieungh** [reply]
Good question, Ryan. I'm not too familiar with the biogeochemistry of grazing in terms of what animals poo out again on the vegetation. But, for this simplistic representation, I think the most important effect on the vegetation is removal of biomass from leaves. One effect of that, in turn, is delayed soil build up and overall poorer nutrient status. Again this is not something I'm very familiar with but as I understand it these two effects are the major influences on vegetation in the ecosystems I'm interested in. (alpine grasslands in Norway)

**Reply by rgknox** [reply]
Lots of cool things we could explore at the interface of grazing and plant chemistry.  I'm thinking that these grazing mechanics could also be used in herbivory of natural systems too.  Plant defence against herbivory, and strategies associated with defense spending would be neat.  We could also explore dynamics between re-absorption of nutrients from leaves during planned scenecense and losses of nutrients through grazed leaves.

**Reply by danicalombardozzi** [reply]
I wonder if it makes sense to eventually connect 'emissions' to the FAN model (see Fan 2019 ctsm #767) that Peter Hess's group at Cornell is working on. 

Excited to see all the various threads on this topic!

**Comment by glemieux** [comment]
Note: we've converted an old issue on dynamic land units to a new discussion topic: https://github.com/NGEET/fates/discussions/943?converting=1.  Lets split this discussion and focus this topic to pasture-only.

**Comment by glemieux** [comment]
> a) the different land uses would all be on the same column and share the same soil and litter. This limits how we can look at the impacts of land use on column based processes including nutrients, water, snow and fire.
>
> b) we don't currently have the full set of LUH2 transitions read into FATES (or either HLM, right?) so we can't partition the gross changes appropriately. (I am a bit hazy, to be honest, on how the existing scheme will differ from LUH2-land. If someone else has a better way of articulating this that would be useful).

Note that the second item here has been address via #1040 and #1116 (and #1239).  Grazing will start to be addressed via #1140.

**Comment by ckoven** [comment]
This is now a thing, with #1140 merged into main. Closing.
